### PR TITLE
Add tests for deploying Karmada services using the configuration file with the karmadactl init --config command.

### PR DESCRIPTION
**Pull request by tiansuo114**
**What type of PR is this?**

**What this PR does / why we need it**:
This PR is an additional test, based on the suggestion provided in [url](https://github.com/karmada-io/karmada/pull/5357#discussion_r1807568543).
**Which issue(s) this PR fixes**:
Part of [#5357](https://github.com/karmada-io/karmada/pull/5357)
It adds test cases for deploying Karmada services using the `karmadactl init --config` command with a configuration file, similar to the existing tests for `karmadactl init`.

**Special notes for your reviewer**:
@liangyuanpeng 
@RainbowMango 

~~This PR currently contains two commits because the test files depend on the new feature for deploying with a configuration file, introduced in [#5357](https://github.com/karmada-io/karmada/pull/5357). Once that PR is merged, this PR will be rebased, retaining only the commit for adding test cases. However, to save time, it is being submitted for review in advance.~~

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by liangyuanpeng** [comment]
Hi, this is a bot comment, just put the label of `ok-to-test` here, you can comment `/retest` to rerun github workflow if github workflow is failing and it's not releated with this PR. 

/ok-to-test

**Comment by tiansuo114** [comment]
Since the [PR](https://github.com/karmada-io/karmada/pull/5357) for adding features has been merged, the commit history in this PR is now normal, so we can proceed with the review and merging correctly.

@liangyuanpeng @RainbowMango

**Review by RainbowMango — COMMENTED** [review]
/assign

**Comment by RainbowMango** [comment]
Ping @liangyuanpeng to review.

File: hack/cli-testing-init-with-config.sh [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+filled_config=$(echo "$CONFIG_TEMPLATE" | sed \
+  -e "s|\${KUBECONFIG_PATH}|${KUBECONFIG_PATH}|g" \
+  -e "s|\${HOST_CLUSTER_NAME}|${HOST_CLUSTER_NAME}|g" \
+  -e "s|\${REGISTRY}|${REGISTRY}|g" \
+  -e "s|\${VERSION}|${VERSION}|g" \
+  -e "s|\${HOME}|${HOME}|g")
+
+echo "$filled_config" > $CONFIG_FILE_PATH
+
+echo "Karmada init config file generated at $CONFIG_FILE_PATH"
```
Comment (liangyuanpeng) [inline hack/cli-testing-init-with-config.sh:147]:
```suggestion
echo "Karmada init config file generated at ${CONFIG_FILE_PATH}"
```

File: hack/cli-testing-init-with-config.sh [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+  echo "Creating directory ${CONFIG_DIR}..."
+  mkdir -p "${CONFIG_DIR}"
+fi
+
+# build Karmada init configuration file
+CONFIG_TEMPLATE=$(cat <<EOF
+apiVersion: config.karmada.io/v1alpha1
+kind: KarmadaInitConfig
+metadata:
+  name: karmada-init
```
Comment (liangyuanpeng) [inline hack/cli-testing-init-with-config.sh:115]: seems like this is unnecessarily.

**Review by liangyuanpeng — COMMENTED** [review]
/assign

File: hack/cli-testing-init-with-config.sh (hunk 2 above)
Comment (tiansuo114) [reply hack/cli-testing-init-with-config.sh]: My understanding is that the information in the metadata is redundant. In the current karmadactl system, the data in the metadata is indeed unnecessary. I have already removed the related data items and addressed the issues mentioned in other comments. If my understanding is incorrect or if additional issues are identified, I hope you can let me know. I look forward to your further guidance.

**Comment by tiansuo114** [comment]
Based on the OSPP project's requirements, the content of this PR has been moved to this [PR](https://github.com/karmada-io/karmada/pull/5306), and this PR will be closed.

**Comment by tiansuo114** [comment]
Based on the OSPP project's requirements, the content of this PR has been moved to this [pr](https://github.com/karmada-io/karmada/pull/5306), and this PR will be closed.

@liangyuanpeng @RainbowMango

**Pull request closed without merging**